action/protocol/staking/candidate_statemanager.go

```diff
@@ -30,17 +30,16 @@ type (
 		delBucket(index uint64) error
 		putBucketAndIndex(bucket *VoteBucket) (uint64, error)
 		delBucketAndIndex(owner, cand address.Address, index uint64) error
-		putBucketIndex(addr address.Address, prefix byte, index uint64) error
-		delBucketIndex(addr address.Address, prefix byte, index uint64) error
 	}
 	// CandidateSet related to setting candidates
 	CandidateSet interface {
-		putCandidate(d *Candidate) error
-		delCandidate(name address.Address) error
-		putVoterBucketIndex(addr address.Address, index uint64) error
-		delVoterBucketIndex(addr address.Address, index uint64) error
-		putCandBucketIndex(addr address.Address, index uint64) error
-		delCandBucketIndex(addr address.Address, index uint64) error
+		putCandidate(*Candidate) error
+		delCandidate(address.Address) error
+		putCandidateList(CandidateList, []byte) error
```

Comment:
zhi: why `putCandidateList` is a function of interface?

Reply:
b/c we are calling it from the `CandidateStateManager` interface, for the same reason why `putCandidate` is a func of interface here

Reply:
no need to add. They are under the same pkg

Reply:
fixed